src/main/java/com/github/tombentley/klog/snapshot/reader/Snapshot.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tombentley.klog.snapshot.reader;
+
+import com.github.tombentley.klog.snapshot.model.ProducerState;
+import java.util.stream.Stream;
+
+public class Snapshot {
+    public enum Type {
```

Comment:
I see your point (including the type naming, that I'll fix). I decided to create a snapshot type because I see that with kraft we will have another type generated from the __cluster_metadata internal topic (ext: .checkpoint). Probably this is over engineering, but I thought that we may need to implement new types in the future. Make sense?

Reply:
But the record class reflecting a `.checkpoint` shouldn't have a `Stream<ProducerState> states` member, I think?

Reply:
Ok, I'm still not familiar with kraft implementation details.